### ISSUE TITLE
Releases generated from tags in incorrect order

### ISSUE THREAD

**Issue by mmstick** [rustic-games/jilu#10]
Without a template defined, I have a project which generates releases from tags in this order:

```md
- [unreleased](#unreleased)
- [`0.4.0`](#040) – _2019.03.15_
- [`0.3.0`](#030) – _2019.02.06_
- [`0.2.0`](#020) – _2019.01.14_
- [`0.1.0`](#010) – _2019.01.08_
- [`0.5.7`](#057) – _2019.09.13_
- [`0.5.6`](#056) – _2019.09.13_
- [`0.5.5`](#055) – _2019.09.12_
- [`0.5.4`](#054) – _2019.09.09_
- [`0.5.3`](#053) – _2019.07.25_
- [`0.5.2`](#052) – _2019.07.25_
- [`0.5.1`](#051) – _2019.07.24_
- [`0.5.0`](#050) – _2019.07.10_
- [`0.4.8`](#048) – _2019.06.21_
- [`0.4.7`](#047) – _2019.06.20_
- [`0.4.6`](#046) – _2019.06.19_
- [`0.4.5`](#045) – _2019.06.03_
- [`0.4.4`](#044) – _2019.06.03_
- [`0.4.3`](#043) – _2019.05.31_
- [`0.4.2`](#042) – _2019.05.20_
- [`0.4.1`](#041) – _2019.05.14_
```

**Comment by JeanMertz** [comment]
Thanks for reporting. It's interesting how `0.4.1` to `0.5.7` are sorted as expected, and `0.1.0` to `0.4.0` are as well, except that last block should have been placed last.

Is there anything else you can tell me about these tags? Are some of them perhaps annotated tags, and other lightweight tags?

Also, is the project you are talking about open-source, so that I can use it as a test case? If not, no worries, I can create a simple test case locally.

**Comment by JeanMertz** [comment]
I've cut a new [0.2.0 release](https://github.com/rustic-games/jilu/releases/tag/v0.2.0).

I don't think it solves this particular issue, as I haven't been able to reproduce it yet, but it does contain one specific fix (b101f8caa52890b8776a13f5381696e7be2912be) related to assigning the right commits to the right releases.

I can think of two reasons why the behaviour you are seeing is happening:

1. The first four releases (`0.1.0` through `0.4.0`) are from _lightweight_ tags, i.e. tags without any annotations, while the other ones have annotated data?

2. The first four releases were manipulated/edited after the last (`0.5.7`) tag.

I'm still unsure why any of these two situations would cause this problem, but it's still worth knowing if any of the two situations match your case.

I will try to reproduce the error later today.

**Comment by mmstick** [comment]
The order of releases is now correct. Now the only problem remaining is that commits aren't being added to the changelog for any of the releases. This was happening in the prior commit as well.

For example, with the following recent changes between two tags:

```
commit 2ba570cca638765a70eaa0fccf4f44397a9ef74f (HEAD -> master, tag: 0.5.8, origin/master, origin/HEAD)
Author: Michael Aaron Murphy <mmstick@pm.me>
Date:   Tue Sep 17 09:24:32 2019 -0600

    chore(release): Updated version to 0.5.8

commit 597b4b7c92cba5a526e49823a4ff4df5fcb5d320
Author: Michael Aaron Murphy <mmstick@pm.me>
Date:   Tue Sep 17 09:23:17 2019 -0600

    chore(deps): Updated hal-rs

commit 5f00d73aff2e077c32d78778e04babf0d1a0205e
Author: Michael Aaron Murphy <mmstick@pm.me>
Date:   Tue Sep 17 09:23:01 2019 -0600

    fix(gtk): Fixed PO list alignments

commit de35949bf3b1130292df2204d1ca15fcc1ffe326 (tag: 0.5.7)
Author: Michael Aaron Murphy <mmstick@pm.me>
Date:   Mon Sep 16 14:08:11 2019 -0600

    chore(release): Update version to 0.5.7
```

Seems to generate:

```
## _[Unreleased]_

_nothing new to show for… yet!_

## [0.5.8]

_2019.09.17_

### Changes

## [0.5.7]

_2019.09.16_

### Changes
```


The project isn't open source. It's a receiving application for use internally by the company. All tags are created with `git tag -a $version -m ''`. They were created on the dates specified, so they've all been happening in chronological order.

**Comment by JeanMertz** [comment]
I've released a new version `0.3.0` that solves the ordering of tags. Let me know if this solves the problem for you.

**Comment by mmstick** [comment]
Tags are now ordered, but it's still dumping all changelogs into a single tagged release. I have versions 0.1 through 0.4.3, and versions 0.4.5 through 0.8.6 completely empty; yet 0.4.0 has all the changes for all releases.
